avr-hal-generic/src/wdt.rs

```diff
@@ -0,0 +1,141 @@
+//! WDT Implementation
+
+pub use embedded_hal::watchdog;
+
+/// Approximate length of the time-out period before the watchdog provides a system reset.
+///
+/// After enabling the watchdog timer, call [`feed`](trait.Watchdog.html#tymethod.feed) before the
+/// period ends to prevent a reset.
+///
+/// Some AVR models do not support all period lengths.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum WatchdogTimeOutPeriod {
+    /// 16 milliseconds
+    Ms16,
+    /// 32 milliseconds
+    Ms32,
+    /// 64 milliseconds
+    Ms64,
+    /// 125 milliseconds
+    Ms125,
+    /// 250 milliseconds
+    Ms250,
+    /// 500 milliseconds
+    Ms500,
+    /// 1 second
+    Ms1000,
+    /// 2 seconds
+    Ms2000,
+    /// 4 seconds
+    Ms4000,
+    /// 8 seconds
+    Ms8000,
+}
+
+/// Implement traits for the watchdog interface
+#[macro_export]
+macro_rules! impl_wdt {
+    (
+        $(#[$wdt_attr:meta])*
+        pub struct $Wdt:ident {
+            mcu_status_register: $MCUSR:ty,
+            peripheral: $WDT:ty,
+        }
+    ) => {
+        use avr_device::generic::Reg;
+        pub use avr_hal::hal::watchdog::*;
+
+        pub struct $Wdt {
```

Comment:
As far as I can see, the `$wdt_attr` is unused right now.  Add it to the struct like this:

```suggestion
        $(#[$wdt_attr])*
        pub struct $Wdt {
```

_but_, on the other hand, I'm thinking that it might be more sensible to directly document the `$Wdt` struct here because documentation should be the same for all watchdogs.  Or do you see an issue with that?

Reply:
I chose to move the documentation into the macro.